src/Illuminate/Container/Container.php

```diff
@@ -547,26 +547,27 @@ public function factory($abstract)
     }
 
     /**
-     * Resolve the given type with the given parameter overrides.
+     * An alias function name for make().
      *
      * @param  string  $abstract
      * @param  array  $parameters
      * @return mixed
      */
-    public function makeWith($abstract, array $parameters)
+    public function makeWith($abstract, array $parameters = [])
```

Comment:
Definitely do not remove it! I was thinking about annotate as `@deprecated` for future removal.